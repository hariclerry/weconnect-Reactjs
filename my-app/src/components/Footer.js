import React from "react";
import "./static/css/fix-footer.css";

const Footer = () => (
  <div>
    <div className="footer">
      <footer className="row">
        <div className="panel-footer text-center bg-dark">
          <p>
            {" "}
            &copy; Copyright 2018 <i>powered by</i> <b> InfoClan</b>{" "}
          </p>
        </div>
      </footer>
    </div>
  </div>
);

export default Footer;
