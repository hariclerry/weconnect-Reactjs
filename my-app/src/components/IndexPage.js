import React from 'react';
import './static/css/bootstrap.min.css'
import './static/css/jumbotron.css'

export default class Index extends React.Component {
    render () {
        return (
            <div>
                     
  
  <nav className="navbar navbar-inverse">
    <div className="container-fluid">
      <div className="navbar-header">
        <a className="navbar-brand" href="help.html">WeConnect</a>
      </div>
      <ul className="nav navbar-nav">
        <li className="active"><a href="index.html">Home</a></li>
        <li><a href="dashboard.html">Dashboard</a></li>
        <li><a href="help.html">Help</a></li>
      </ul>
      <ul className="nav navbar-nav navbar-right">
    
        <li><a href="/signup"><span className="glyphicon glyphicon-user"></span>Sign Up</a></li>
        <li><a href="/login"><span className="glyphicon glyphicon-log-in"></span>Sign In</a></li>
        
      </ul>
    </div>
  </nav> 

<div className="jumbotron">
          <div className="container">
           <h1>We Connect </h1>
                
           </div>
 </div>
          
            
<br/>
    <div className="container">
    <h3> Welcome to We Connect App, where you can Register, Give Review and Connect with millions of Businesses World wide!</h3>
    <a href="signup.html" className="btn btn-start" role="button"><h4>Get Started</h4></a>
    </div>
    <hr/>
    
    <div className="container">
    <div className="row">
      <div className="col-sm-6 col-md-4">
        <div className="thumbnail">
          <img src="static/images/reg.jpeg" alt="Register Business" />
  
          <div className="caption">
            <h3>Register Business</h3>
            <p>Lorem ipsum donec id elit non mi porta gravida at eget metus.</p>
          </div>
        </div>
      </div>
    
      <div className="col-sm-6 col-md-4">
          <div className="thumbnail">
            <img src="static/images/innovation.jpeg" alt="Give Review" />
            <div className="caption">
              <h3>Give Review</h3>
              <p>Lorem ipsum donec id elit non mi porta gravida at eget metus.</p>
            </div>
          </div>
        </div>
    
        <div className="col-sm-6 col-md-4">
            <div className="thumbnail">
              <img src="static/images/business.jpeg"  alt="Connect" />
             
              <div className="caption">
                <h3>Connect</h3>
                <p>Lorem ipsum donec id elit non mi porta gravida at eget metus.</p>
              </div>
            </div>
          </div>
      </div>
    </div>


   

   <br/>
   <br/>
   <br/>
   <br/>
    <footer className="row">
           
            <div className="panel-footer text-center bg-dark">
        <p>  &copy; Copyright 2018 <i>powered by</i>  <b> InfoClan</b> </p>
            </div>
    
    </footer>     

            </div>

        )
    }
}