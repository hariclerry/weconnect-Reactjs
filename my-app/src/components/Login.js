import React from "react";

// import { NotificationManager } from 'react-notifications';

import Footer from "./footer";
import AuthNavbar from "./authNavbar"
import "./static/css/fix-footer.css";
import "./static/css/bootstrap.min.css";
import "./static/css/stylesform.css";

const Login = ({ Login }) => (
  <div>
    <AuthNavbar />
    <div className="container">
      <h3> Please sign in to do more</h3>
    </div>
    <hr />
    <div className="container" id="log-in-form">
      <div className="heading">
        <h1 style={{ textShadow: "0 0 0 black", color: "#000000" }}>
          Log In Your Account
        </h1>
      </div>
      <form onSubmit={Login}>
        <div className="form-group">
          <label htmlFor="email">Email:</label>
          <input
            type="email"
            className="form-control"
            name="email"
            id="email"
            placeholder="Enter email"
          />
        </div>
        <div className="form-group">
          <label htmlFor="pwd">Password:</label>
          <input
            type="password"
            className="form-control"
            name="password"
            id="pwd"
            placeholder="Enter password"
          />
        </div>
        <div className="form-group form-group-btn">
          <button
            type="submit"
            className="btn btn-success btn-lg"
            style={{ marginTop: "20%" }}
          >
            Log In
          </button>
        </div>
        <div className="clearfix" />
      </form>
    </div>
    <Footer />
  </div>
);

export default Login;
