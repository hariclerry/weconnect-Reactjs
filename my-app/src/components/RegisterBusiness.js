import React from 'react';

import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';

// import 'react-notifications/lib/notifications.css';

import { NotificationManager } from 'react-notifications';
import './static/css/bootstrap.min.css'
import { registerBusiness} from '../actions/businessActions';
import Footer from './Footer';


class RegisterBusiness extends React.Component {

    componentWillReceiveProps(recieved) {
        if (recieved && recieved.business.message === "Business successfully registered") {
            this.props.history.push('/dashboard');
        }
        else{
            if(recieved && recieved.business.status === "failure"){
                NotificationManager.error(recieved.business.message, "", 5000);
            }
        }
};

	jsonStringify = object =>{
        let simpleObj={};
            for (let prop in object){
                    if (!object.hasOwnProperty(prop)){
                            continue;
                        }
                    if (typeof(object[prop]) === 'object'){
                            continue;
                        }
                    simpleObj[prop] = object[prop];
                }
                return JSON.stringify(simpleObj)}

    createBusiness = e => {
        e.preventDefault(); // prevent the default link behavior of opening a new page
        let businessCredential = {
            name: e.target.elements.name.value,
            category: e.target.elements.category.value,
            location: e.target.elements.location.value,
            description: e.target.elements.description.value
        };
        this.props.registerBusiness(this.jsonStringify(businessCredential))

    }
    render () {
        return (
            <div>
                 <nav className="navbar navbar-inverse">
    <div className="container-fluid">
      <div className="navbar-header">
        <a className="navbar-brand" href="/">WeConnect</a>
      </div>
     
      
        <ul className="nav navbar-nav navbar-right">
            <li><a href=""><span className="glyphicon glyphicon-user"></span>User</a></li>
            <li><a href="index.html"><span className="glyphicon glyphicon-log-out"></span> Logout</a></li>
              </ul>
       <ul className="nav navbar-nav">
       <li><a href="/">Home</a></li>
            <li>
                <a href="/dashboard">Dasboard</a>
            </li>
            <li>
                <a href="/registerbusiness">Register Business</a>
            </li>
            <li>
                <a href="/businesslist">All Businesses</a>
            </li>
            <li>
                <a href="/singlebusiness">Single Business</a>
            </li>
        </ul>
  </div>
  </nav>
    
             
                  

                        <div className="container">
                               <h3> You can now Register your Business</h3>
                        </div>
                        <hr/>
                        <div style={{margin: "100px"}}>
                        <form onSubmit={this.createBusiness }>
                          
                            <div className="form-group"> 
                              <label className="control-label requiredField" htmlFor="title">Business Name:</label>
                              <input className="form-control"  name="name" id="title" type="text"/>
                            </div>

                            <div className="form-group">
                                    <label className="control-label " htmlFor="category">Business Category:</label>
                                    <input className="form-control"  id="category" name="category" type="text"/>
                            </div>
                            
                            <div className="form-group">
                              <label className="control-label " htmlFor="location">Business Location:</label>
                              <input className="form-control"  id="location" name="location" type="text"/>
                            </div>
                            
                            
                            <div className="form-group"> 
                              <label className="control-label " htmlFor="description">Business Description:</label>
                              <textarea className="form-control" cols="40" id="description" name="description" rows="10"></textarea>
                            </div>
                            
                            <div className="form-group">
                              <button className="btn btn-primary " name="submit" type="submit">Create</button>
                            </div>
                            
                          </form>	
                          </div>						
                                      
                       
                        
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <br/>
    <Footer />

    </div>

        )
    }
}


RegisterBusiness.protoTypes = {
    registerBusiness: PropTypes.func.isRequired
}

const mapStateToProps = state =>({
    business: state.business.createBusinessMessage

});

export default  withRouter(connect(mapStateToProps,{ registerBusiness })(RegisterBusiness));