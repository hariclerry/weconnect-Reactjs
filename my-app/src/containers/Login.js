import React from "react";

import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
// import { NotificationManager } from 'react-notifications';

import { loginUser } from "../actions/userActions";
import { jsonStringify } from "../utils/jsonHelper";
import Login from "../components/login";


class UserLogin extends React.Component {
  //     componentWillReceiveProps(recieved) {
  //         console.log(recieved)
  //         if (recieved && recieved.user.message === "You logged in successfully.") {
  //             this.props.history.push('/dashboard');
  //         }
  //         else{
  //             if(recieved && recieved.user.status === "failure"){
  //                 NotificationManager.error(recieved.user.message, "", 5000);
  //             }
  //         }
  // };


  Login = e => {
    e.preventDefault(); //o prevent the default link behavior of opening a new page
    let userCredential = {
      email: e.target.elements.email.value,
      password: e.target.elements.password.value
    };
    this.props
      .loginUser(jsonStringify(userCredential))
      .then(() => {
        this.props.history.push("/dashboard");
      })
      .catch(err => {
        console.log(err.data);
        alert("Wrong email or password");
      });
  };
  render() {

    return (
      <div>
        <Login 
        Login={this.Login}/>
      </div>
    );
  }
}

Login.protoTypes = {
  loginUser: PropTypes.func.isRequired,
  user: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  user: state.auth.loginData
});

export default withRouter(
  connect(
    mapStateToProps,
    { loginUser }
  )(UserLogin)
);
