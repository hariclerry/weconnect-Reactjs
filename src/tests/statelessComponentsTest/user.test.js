import React from "react";
import {shallow} from "enzyme";
import RegisterBusiness from "components/business/registerBusiness";
// import Businesses from "components/business/businessList";
import UserRegistration from "components/user/signup";
import Login from "components/user/login";
import ResetUserPassword from "components/user/resetPassword";
import AuthNav from "components/common/authNavbar";
import Footer from "components/common/footer";
import IndexPage from "components/common/indexPage";



describe('UserRegistration component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<UserRegistration />)
      expect(wrapper).toMatchSnapshot();
  })
})


describe('Login component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<Login />)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('ResetUserPassword component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<ResetUserPassword />)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('AuthNav component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<AuthNav />)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('Footer component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<Footer />)
      expect(wrapper).toMatchSnapshot();
  })
})

describe('IndexPage component',() => {

  it('should match snapshot',() =>{
      const wrapper = shallow(<IndexPage />)
      expect(wrapper).toMatchSnapshot();
  })
})

