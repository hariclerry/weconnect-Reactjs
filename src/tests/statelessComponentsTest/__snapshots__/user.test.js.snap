// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AuthNav component should match snapshot 1`] = `
<nav
  className="navbar navbar-inverse"
>
  <div
    className="container-fluid"
  >
    <div
      className="navbar-header"
    >
      <a
        className="navbar-brand"
        href="/"
      >
        WeConnect
      </a>
    </div>
    <ul
      className="nav navbar-nav"
    >
      <li
        className="active"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          Home
        </NavLink>
      </li>
    </ul>
    <ul
      className="nav navbar-nav navbar-right"
    >
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/signup"
        >
          <span
            className="glyphicon glyphicon-user"
          />
           Sign Up
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/login"
        >
          <span
            className="glyphicon glyphicon-log-in"
          />
           Sign In
        </NavLink>
      </li>
    </ul>
  </div>
</nav>
`;

exports[`Footer component should match snapshot 1`] = `
<div>
  <div
    className="footer"
  >
    <footer
      className="container-fluid text-center"
      data-offset-bottom="10"
    >
      <div
        className="panel-footer text-center bg-dark"
      >
        <p>
           
          Â© Copyright 2018 
          <i>
            powered by
          </i>
           
          <b>
             InfoClan
          </b>
           
        </p>
      </div>
    </footer>
  </div>
</div>
`;

exports[`IndexPage component should match snapshot 1`] = `
<div>
  <AuthNav />
  <div
    className="jumbotron"
  >
    <div
      className="container"
    >
      <h1>
        We Connect 
      </h1>
    </div>
  </div>
  <br />
  <div
    className="container"
  >
    <h3>
       
      Welcome to We Connect! Connect with millions of Businesses World wide
    </h3>
    <a
      className="btn btn-start"
      href="/signup"
      role="button"
    >
      <h4>
        Get Started
      </h4>
    </a>
  </div>
  <hr />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm-6 col-md-4"
      >
        <div
          className="thumbnail"
        >
          <img
            alt="image1"
            src="reg.jpeg"
          />
          <div
            className="caption"
          >
            <h3>
              Register Business
            </h3>
            <p>
              Lorem ipsum donec id elit non mi porta gravida at eget metus.
            </p>
          </div>
        </div>
      </div>
      <div
        className="col-sm-6 col-md-4"
      >
        <div
          className="thumbnail"
        >
          <img
            alt="image2"
            src="innovation.jpeg"
          />
          <div
            className="caption"
          >
            <h3>
              Give Review
            </h3>
            <p>
              Lorem ipsum donec id elit non mi porta gravida at eget metus.
            </p>
          </div>
        </div>
      </div>
      <div
        className="col-sm-6 col-md-4"
      >
        <div
          className="thumbnail"
        >
          <img
            alt="image3"
            src="business.jpeg"
          />
          <div
            className="caption"
          >
            <h3>
              Connect
            </h3>
            <p>
              Lorem ipsum donec id elit non mi porta gravida at eget metus.
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <Footer />
</div>
`;

exports[`Login component should match snapshot 1`] = `
<div>
  <_class
    options={Object {}}
  />
  <AuthNav />
  <div
    className="container"
  >
    <h3>
       Please sign in to do more
    </h3>
  </div>
  <hr />
  <div
    className="container"
    id="log-in-form"
  >
    <div
      className="heading"
    >
      <h1
        style={
          Object {
            "color": "#000000",
            "textShadow": "0 0 0 black",
          }
        }
      >
        Log In Your Account
      </h1>
    </div>
    <form>
      <div
        className="form-group"
      >
        <label
          htmlFor="email"
        >
          Email:
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          placeholder="Enter email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="pwd"
        >
          Password:
        </label>
        <input
          className="form-control"
          id="pwd"
          name="password"
          placeholder="Enter password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group form-group-btn"
      >
        <button
          className="btn btn-success btn-lg"
          style={
            Object {
              "marginTop": "20%",
            }
          }
          type="submit"
        >
          Log In
        </button>
      </div>
      <div
        className="clearfix"
      />
    </form>
  </div>
  <Footer />
</div>
`;

exports[`ResetUserPassword component should match snapshot 1`] = `
<div>
  <withRouter(Connect(NavBar)) />
  <div
    className="container"
  >
    <h3>
       You can now reset your password
    </h3>
  </div>
  <hr />
  <div
    style={
      Object {
        "margin": "200px 400px",
      }
    }
  >
    <form>
      <div
        className="form-group"
      >
        <label
          htmlFor="email"
        >
          Email:
        </label>
        <input
          className="form-control"
          id="email"
          name="email"
          placeholder="Enter email"
          required={true}
          type="email"
        />
      </div>
      <div
        className="form-group"
      >
        <label
          htmlFor="pwd"
        >
           New Password:
        </label>
        <input
          className="form-control"
          id="pwd"
          name="new_password"
          placeholder="Enter password"
          required={true}
          type="password"
        />
      </div>
      <div
        className="form-group form-group-btn"
      >
        <button
          className="btn btn-success btn-lg"
          style={
            Object {
              "marginTop": "3%",
            }
          }
          type="submit"
        >
          Reset Password
        </button>
      </div>
      <div
        className="clearfix"
      />
    </form>
  </div>
  <Footer />
</div>
`;

exports[`UserRegistration component should match snapshot 1`] = `
<div>
  <_class
    options={
      Object {
        "top": "50px",
        "zIndex": 200,
      }
    }
  />
  <AuthNav />
  <div
    className="container"
  >
    <h3>
       Please register to login
    </h3>
  </div>
  <hr />
  <div
    className="container"
    id="registration-form"
  >
    <div
      className="image"
    />
    <div
      className="frm"
    >
      <h1>
        Sign up
      </h1>
      <form>
        <div
          className="form-group"
        >
          <label
            htmlFor="username"
          >
            Username:
          </label>
          <input
            className="form-control"
            id="username"
            name="name"
            placeholder="Enter username"
            required={true}
            type="text"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="email"
          >
            Email:
          </label>
          <input
            className="form-control"
            id="email"
            name="email"
            placeholder="Enter email"
            required={true}
            type="email"
          />
        </div>
        <div
          className="form-group"
        >
          <label
            htmlFor="pwd"
          >
            Password:
          </label>
          <input
            className="form-control"
            id="pwd"
            name="password"
            placeholder="Enter password"
            required={true}
            type="password"
          />
        </div>
        <div
          className="fixed"
        >
          <button
            className="btn btn-success btn-lg"
            type="submit"
          >
            Submit
          </button>
        </div>
      </form>
    </div>
  </div>
  <Footer />
</div>
`;
